components/sidebar/sidebar_service.cc

```diff
@@ -393,23 +473,29 @@ void SidebarService::LoadSidebarItems() {
       }
       // Always use latest properties for built-in type item.
       if (type == SidebarItem::Type::kTypeBuiltIn) {
-        if (const auto value =
-                item.FindIntKey(kSidebarItemBuiltInItemTypeKey)) {
-          auto id = static_cast<SidebarItem::BuiltInItemType>(*value);
-          auto iter = std::find_if(
-              default_items_to_add.begin(), default_items_to_add.end(),
-              [id](const auto& default_item) {
-                return default_item.built_in_item_type == id;
-              });
-          // It might be an item which is no longer is offered as built-in
-          if (iter == default_items_to_add.end()) {
-            continue;
-          }
-          // Valid built-in item, add it
-          items_.emplace_back(*std::make_move_iterator(iter));
-          default_items_to_add.erase(iter);
+        const auto built_in_type_value =
+            item.FindIntKey(kSidebarItemBuiltInItemTypeKey);
+        if (!built_in_type_value.has_value()) {
+          VLOG(1) << "built-in item did not have a type: "
+                  << item.DebugString();
           continue;
```

Comment:
These lines solve the issue since we should skip parsing the item if we have bad data. The previous version of this line was missing the `continue`. Plus it isn't immediately obvious that:
```
if (const auto value =
                item.FindIntKey(kSidebarItemBuiltInItemTypeKey)
```
is the same as:
```
        const auto value = item.FindIntKey(kSidebarItemBuiltInItemTypeKey);
        if (!value.has_value()) 
```